module_analysis/1_calc_module_composition.py

```diff
@@ -0,0 +1,283 @@
+        # Should I do 2 one-sided tests, one for testing if
+        odds_ratio, pval = scipy.stats.fisher_exact(observed_contingency_table)
+
+        # odds_ratio, pval_greater = scipy.stats.fisher_exact(
+        #    observed_contingency_table, alternative="greater"
+        # )
+        # odds_ratio, pval_less = scipy.stats.fisher_exact(
+        #    observed_contingency_table, alternative="less"
+        # )
+
+        # Fill in df
+        out_df.loc[module_id, "num core"] = len(core_genes_in_module)
+        out_df.loc[module_id, "num acc"] = len(acc_genes_in_module)
+        out_df.loc[module_id, "odds ratio"] = odds_ratio
+        out_df.loc[module_id, "p-value"] = pval
+
+        # Do we want to include p-value criteria?
+        # Given the small sizes of the modules I think this is why
+        # most of the findings are not significant
+        # Use bonferrroni corrected p-value
+        if odds_ratio > 1 and pval < 0.05 / len(all_genes):
+            out_df.loc[module_id, "module label"] = "mostly core"
+        elif odds_ratio < 1 and pval < 0.05 / len(all_genes):
+            out_df.loc[module_id, "module label"] = "mostly accessory"
+        else:
+            out_df.loc[module_id, "module label"] = "mixed"
+
+        # If using two separate one-sided tests
+        # if odds_ratio > 1 and pval_greater < 0.05:
+        #    out_df.loc[module_id, "module label"] = "mostly core"
+        # elif odds_ratio < 1 and pval_less < 0.05:
+        #    out_df.loc[module_id, "module label"] = "mostly accessory"
+        # else:
+        #    out_df.loc[module_id, "module label"] = "mixed"
+
+    return out_df
+
+
+# %%time
+# Get labels for PAO1 compendium
+pao1_module_labels = label_modules(
+    pao1_module_ids, pao1_membership, pao1_core, pao1_acc, pao1_gene_group_composition
+)
+
+# %%time
+# Get labels of PA14 compendium
+pa14_module_labels = label_modules(
+    pa14_module_ids, pa14_membership, pa14_core, pa14_acc, pa14_gene_group_composition
+)
+
+pao1_module_labels.head()
+
+pa14_module_labels.head()
+
+# Distribution of p-values
+sns.displot(pao1_module_labels["p-value"])
+sns.displot(pa14_module_labels["p-value"])
+
+# ### Examine module composition
+
+pao1_module_labels["module label"].value_counts()
+
+pa14_module_labels["module label"].value_counts()
+
+# Add size of modules to df
+pao1_module_labels["size"] = pao1_membership["module id"].value_counts()
+pa14_module_labels["size"] = pa14_membership["module id"].value_counts()
+
+pao1_module_labels.head()
+
+pa14_module_labels.head()
+
+# +
+# Size distributions of PAO1 modules
+f1 = sns.displot(
+    pao1_module_labels.loc[pao1_module_labels["module label"] == "mixed", "size"]
+)
+plt.title("Size distribution of PAO1 mixed modules")
+
+f2 = sns.displot(
+    pao1_module_labels.loc[
+        pao1_module_labels["module label"] == "mostly accessory", "size"
+    ]
+)
+plt.title("Size distribution of PAO1 mostly accessory modules")
+
+f3 = sns.displot(
```

Comment:
Is this suppose to be a blank plot? It showed up for me as blank and I want to confirm it isn't a bug.

Reply:
Yes, thats correct. Since there are 0 modules labeled as mostly core, this plot is blank. I just have it plotting as I'll fiddle with different upstream methods and will just want to re-run this notebook